File: app/src/main/java/com/softaai/designpatterns/java/creational/AbstractFactory/AbstractFactory.java
package com.softaai.designpatterns.java.creational.AbstractFactory;

import com.softaai.designpatterns.java.creational.AbstractFactory.color.Color;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.Shape;

/**
 * Created by Amol Pawar on 31-03-2019.
 * softAai Apps
 */
public abstract class AbstractFactory {
    protected abstract Color getColor(String color);
    protected abstract Shape getShape(String shape);
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/AbstractFactory/AbstractFactory.kt
package com.softaai.designpatterns.kotlin.creational.AbstractFactory

import com.softaai.designpatterns.kotlin.creational.AbstractFactory.color.Color
import com.softaai.designpatterns.kotlin.creational.AbstractFactory.shape.Shape


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */

abstract class AbstractFactory{
    abstract fun getColor(color:String) : Color?
    abstract fun getShape(shape:String) : Shape?
}